Library/Homebrew/extend/pathname.rb

```diff
@@ -308,7 +308,7 @@ def compression_type
   # @private
   def elf?
     @which_file ||= which("file")
-    !@which_file.nil? && `#{@which_file} -b #{self}` =~ /^ELF/
+    !@which_file.nil? && `#{@which_file} -b #{self.to_s.gsub(/(\(|\))/,'\\\\\1')}` =~ /^ELF/
```

Comment:
Use `popen_read` instead of back ticks:
```ruby
Utils.popen_read(@which_file, "-b", self).start_with? "ELF")
```
This method should throw an error if `@which_file` is `nil`